src/v/storage/index_state.cc

```diff
@@ -124,4 +128,90 @@ iobuf index_state::checksum_and_serialize() {
     return serde_compat::index_state_serde::encode(*this);
 }
 
+void index_state::serde_write(iobuf& out) const {
+    using serde::write;
+
+    iobuf tmp;
+    write(tmp, bitflags);
+    write(tmp, base_offset);
+    write(tmp, max_offset);
+    write(tmp, base_timestamp);
+    write(tmp, max_timestamp);
+    write(tmp, relative_offset_index.copy());
+    write(tmp, relative_time_index.copy());
+    write(tmp, position_index.copy());
```

Comment:
@felixguendling let's discuss a version of `write` that takes an l-value reference so we can avoid the copy here.

Reply:
Working on it. Seems like `write(T&&)` instead of `write(T)` works but I need to write more tests.

Reply:
https://github.com/vectorizedio/redpanda/issues/3216